server/routes/v1/deployment.ts

```diff
@@ -161,3 +165,59 @@ export const resetDeploymentApprovals = [
     return res.json(deployment)
   },
 ]
+
+export const fetchRawModelFiles = [
+  ensureUserRole('user'),
+  bodyParser.json(),
+  async (req: Request, res: Response) => {
+    const { uuid, version, fileType } = req.params
+    const deployment: any = await findDeploymentByUuid(req.user!, uuid)
+
+    if (!req.user._id.equals(deployment.owner)) {
+      throw Unauthorised(
+        {},
+        `User is not authorised to download this file. Requester: ${req.user._id}, owner: ${deployment.owner}`
+      )
+    }
+
+    if (fileType !== 'code' && fileType !== 'binary') {
+      throw NotFound({ fileType }, 'Unknown file type specificed')
+    }
+
+    if (deployment === null) {
+      throw NotFound({ deploymentUuid: uuid }, `Unable to find deployment for uuid ${uuid}`)
+    }
+
+    const versionDocument: any = await findVersionByName(req.user!, deployment.model, version)
+    const bucketName: string = config.get('minio.uploadBucket')
+    const client = new Minio.Client(config.get('minio'))
+
+    let filePath
+
+    if (fileType === 'code') {
+      filePath = versionDocument.rawCodePath
+    }
+    if (fileType === 'binary') {
+      filePath = versionDocument.rawBinaryPath
+    }
+
+    // Stat object to get size so browser can determine progress
+    const { size } = await client.statObject(bucketName, filePath)
+
+    res.set('Content-Disposition', contentDisposition(fileType, { type: 'inline' }))
```

Comment:
This is not the filetype but the file name

Reply:
I was using the file type as the name, so when it's download it'll be "code.zip" or "binary.zip". I assumed they'd be better than an objectId but happy to change